static/js/components/CommentList.jsx

```diff
@@ -39,6 +170,9 @@ const CommentList = () => {
   const obj = useSelector((state) => state.source);
   const userProfile = useSelector((state) => state.profile);
   const permissions = useSelector((state) => state.profile.permissions);
+  const compactComments = useSelector(
+    (state) => state.profile.preferences.compactedComments
```

Comment:
```suggestion
    (state) => state.profile.preferences?.compactComments
```
Just in case `preferences` is uninitialized, and `compacted` -> `compact` for consistency